pandas/core/resample.py

```diff
@@ -1389,9 +1389,15 @@ def __init__(
                 label = "right"
         else:
             if closed is None:
-                closed = "left"
+                if origin in ["end", "end_day"]:
```

Comment:
could check `if origin in ["end", "end_day"]` at the top and do `closed is None`/`label is None` inside